src/components/Filter/components/predefinedFilter.js

```diff
@@ -1,33 +1,56 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect, useCallback } from 'react';
 
 type OptionShape = {
   text: string,
   value: string,
 };
 
 type PredefinedFilterProps = {
-  dropdownRef: any,
-  onToggle: Function,
   options: OptionShape[],
   onSelect: Function,
   baseClass?: string,
   filterButtonContent?: Function,
 };
 
 const PredefinedFilter = ({
-  dropdownRef,
-  onToggle,
   options,
   onSelect,
   baseClass,
   filterButtonContent,
 }: PredefinedFilterProps) => {
   const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef(null);
+
+  const listenerAction = useCallback(() => {
+    setIsOpen(false);
+  }, [setIsOpen]);
+
+  const killListener = useCallback(
+    (): any => document.removeEventListener('click', listenerAction),
+    [listenerAction],
+  );
+
+  const startListener = useCallback(
+    (): any => document.addEventListener('click', listenerAction),
+    [listenerAction],
+  );
+
+  useEffect(() => {
+    if (isOpen) {
+      startListener();
+    } else {
```

Comment:
i don't think this else does anything, when the popup is closed the listener will be closed because of the use effect callback, and if the popup is closed when the component mounts there's no need to remove a listener that hasn't been added (and then removing it again once the effect runs again)

Reply:
when unmounting => killListener
when isOpen != true => killListener (note that on mount this will run but is harmless)
when isOpen == true => startListener

Reply:
>  when isOpen != true => killListener (note that on mount this will run but is harmless)

Exactly though, this code isn't actually doing anything because after the first render we will anyways kill the listener, this will only ever run once the listener has been killed already (with the exception being the first render, then nothing will happen if isOpen is not true)

Reply:
i'm not entirely sure of your argument rob

Reply:
forgot how useEffect actually works vs how i'd like it work